splash/lua_modules/splash.lua

```diff
@@ -98,7 +98,7 @@ local Splash = {}
 Splash.__index = Splash
 
 function Splash.private_create(py_splash)
-  local self = {args=py_splash.args}
+  local self = {args=py_splash.args, js_enabled=py_splash.js_enabled}
```

Comment:
Here you're creating spalsh.js_enabled variable by copying py_splash.js_enabled value to it. So changes to splash.js_enabled won't affect py_splash.js_enabled. `args` is different because it is a dict.